Gems/ROS2/Code/CMakeLists.txt

```
@@ -42,6 +42,18 @@ add_custom_target(
     COMMAND ${CMAKE_COMMAND} -DROS_DISTRO=${ROS_DISTRO} -P ${CMAKE_CURRENT_SOURCE_DIR}/checkROS2Distribution.cmake
 )
 
+# Gazebo messages are optional, so we will only add the dependants only if the package is found.
+# gazebo_msgs package is EOL and will be removed in ROS 2 Kilted Kaiju.
+# If you need to use ContactSensor and ROS2 Spawner, please consider building gazebo_msgs from source.
+find_package(gazebo_msgs QUIET)
+if (gazebo_msgs_FOUND)
+    message(STATUS "Found gazebo_msgs package, enabling legacy features like ContactSensor Component and ROS2 Spawner Component")
+    SET (WITH_GAZEBZOMSGS TRUE)
```

There is a typo: "WITH_GAZEB**Z**OMSGS". I also suggest changing it further, to "WITH_GAZEBO_MSGS".
```suggestion
    SET (WITH_GAZEBO_MSGS TRUE)
```